integ-test/src/test/java/org/opensearch/sql/legacy/AggregationIT.java

```diff
@@ -146,144 +147,104 @@ public void extendedStatsTest() throws IOException {
   }
 
   @Test
-  public void percentileTest() throws IOException {
-
+  public void percentileTest() {
     JSONObject result =
-        executeQuery(String.format("SELECT PERCENTILES(age) FROM %s", TEST_INDEX_ACCOUNT));
-    Assert.assertThat(getTotalHits(result), equalTo(1000));
-    Assert
-        .assertEquals(20.0, getDoubleAggregationValue(result, "PERCENTILES(age)", "values", "1.0"),
-            0.001);
-    Assert
-        .assertEquals(21.0, getDoubleAggregationValue(result, "PERCENTILES(age)", "values", "5.0"),
-            0.001);
-    Assert
-        .assertEquals(25.0, getDoubleAggregationValue(result, "PERCENTILES(age)", "values", "25.0"),
-            0.001);
-    // All percentiles are approximations calculated by t-digest, however, P50 has the widest distribution (not sure why)
-    Assert
-        .assertEquals(30.5, getDoubleAggregationValue(result, "PERCENTILES(age)", "values", "50.0"),
-            0.6);
-    Assert
-        .assertEquals(35.0, getDoubleAggregationValue(result, "PERCENTILES(age)", "values", "75.0"),
-            0.6);
-    Assert
-        .assertEquals(39.0, getDoubleAggregationValue(result, "PERCENTILES(age)", "values", "95.0"),
-            0.6);
-    Assert
-        .assertEquals(40.0, getDoubleAggregationValue(result, "PERCENTILES(age)", "values", "99.0"),
-            0.6);
+        executeJdbcRequest(String.format("SELECT PERCENTILES(age) FROM %s", TEST_INDEX_ACCOUNT));
+    assertEquals(1, result.getInt("total"));
+    verifySchema(result, schema("PERCENTILES(age)", null, "double"));
+    JSONObject jsonObject = result.getJSONArray("datarows").getJSONArray(0).getJSONObject(0);
+    assertTrue(jsonObject.keySet().contains("5.0"));
+    assertTrue(jsonObject.keySet().contains("25.0"));
+    assertTrue(jsonObject.keySet().contains("50.0"));
+    assertTrue(jsonObject.keySet().contains("75.0"));
+    assertTrue(jsonObject.keySet().contains("95.0"));
+    assertTrue(jsonObject.keySet().contains("99.0"));
+    assertTrue(jsonObject.keySet().contains("1.0"));
```

Comment:
Not too sure, but `assertAll` might be able to be used here?
```suggestion
JSONObject jsonObject = result.getJSONArray("datarows").getJSONArray(0).getJSONObject(0);
assertAll(
    () -> assertEquals(1, result.getInt("total")),
    () -> verifySchema(result, schema("PERCENTILES(age)", null, "double")),
    () -> assertTrue(jsonObject.keySet().contains("5.0")),
    () -> assertTrue(jsonObject.keySet().contains("25.0")),
    () -> assertTrue(jsonObject.keySet().contains("50.0")),
    () -> assertTrue(jsonObject.keySet().contains("75.0")),
    () -> assertTrue(jsonObject.keySet().contains("95.0")),
    () -> assertTrue(jsonObject.keySet().contains("99.0"));
    () -> assertTrue(jsonObject.keySet().contains("1.0"))
    )
```

Reply:
Actually there are a lot of tests which may be able to be refactored like this, but it will likely take a lot of your time and might be considered outside of the scope of this PR.